apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionScopeTests.cs

```diff
@@ -371,7 +371,7 @@ public async Task ParallelWriteAsync()
             string[] keys = new string[] { IdGenerator.FromGuid(), IdGenerator.FromGuid(), IdGenerator.FromGuid() };
             await RetryHelpers.ExecuteWithRetryAsync(async () =>
             {
-                using var scope = new TransactionScope();
+                using var scope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
```

Comment:
What's this change?

Reply:
Hmmm, we cannot change a test without understanding why.
I just ran these locally and they were green, so there might be an issue with your environment or maybe even with some of the code your are adding on this PR (although I really cannot think what).
Bottom line, it's OK to modfied tests when we understand what is happening and it makes sense, but not blindly, as we would very well be letting some bugs in.

Reply:
> TransactionScope doesn't work well with async/await and the recommended fix is to ...

Who says that TransactionScope does not work well with async/await? And who recommendas the fix? I'm just curious.

> We got rid of GetConnection() overload and used GetConnection(Logger logger = null, bool logCommitStats = false) instead.

OK, but why did this change had this side effect?

Sorry for insisting, but I don't yet understand what happened, it might be OK to make this change, but we really need to understand why.
I can take a look later as well if you prefer.

Reply:
Update: I logged the ManagedThreadId after every line in the test, before and after making the changes in `GetConnection` and observed that before making the change the thread ID remains the same till the scope completion. After making the change, the thread ID changes before the scope.Complete() call.  This still doesn't explain why the 'GetConnection' changes triggered it. But I _think_ thread ID may or may not change after async/await call in the flow. This isn't predictable and we have one new async await call in the new `GetConnection`

Reply:
OK, having gone through the linked issue I understand and agree with having to make a change on this tests regardless. It's still not 100% clear why the issues manifested with this change, but that's not relevant. Our tests and production code is using async/await, so we should make this change.

One clarification, `SessionPoolManager.Create` is not calling `CreateClientAsync`, it's only passing it as a Func for it to be the client factory, but it's not being called there at all.

By looking at the `SessionPoolManager` code, though, I wonder if getting rid of the simple `GetConnection` method was a good idea. I think it was, as know we always have a fresh SessionPoolmanager for each test, which means tests are isolated from each other, which I believe is what should be. But still I'll let @jskeet chime in.

Reply:
I don't think I understand quite enough about what's been removed to comment - will chat with Amanda offline and see if that helps. (But yes, having test isolation is something I'm generally in favour of :)